diff --git a/node_modules/react-awesome-modal/lib/effect/fadeInDown.js b/node_modules/react-awesome-modal/lib/effect/fadeInDown.js
index 53ad54c..af6d51e 100644
--- a/node_modules/react-awesome-modal/lib/effect/fadeInDown.js
+++ b/node_modules/react-awesome-modal/lib/effect/fadeInDown.js
@@ -40,7 +40,7 @@ exports.default = {
     },
     'panel': {
         'backgroundColor': '#fff',
-        'borderRadius': '5px',
+        'borderRadius': '0px',
         'boxSizing': 'border-box',
         'boxShadow': '0 2px 4px rgba(0, 0, 0, 0.3)',
         'transform': 'translate3d(0, 0, 0)',
diff --git a/node_modules/react-awesome-modal/lib/effect/fadeInLeft.js b/node_modules/react-awesome-modal/lib/effect/fadeInLeft.js
index d4dec74..a71c1b7 100644
--- a/node_modules/react-awesome-modal/lib/effect/fadeInLeft.js
+++ b/node_modules/react-awesome-modal/lib/effect/fadeInLeft.js
@@ -26,7 +26,7 @@ exports.default = {
         'justifyContent': 'center',
         'alignItems': 'center',
         'zIndex': 10000
-    },
+	},
     'containerHidden': {
         'position': 'fixed',
         'top': 0,
@@ -40,7 +40,7 @@ exports.default = {
     },
     'panel': {
         'backgroundColor': '#fff',
-        'borderRadius': '5px',
+        'borderRadius': '0px',
         'boxSizing': 'border-box',
         'boxShadow': '0 2px 4px rgba(0, 0, 0, 0.3)',
         'transform': 'translate3d(0, 0, 0)',
diff --git a/node_modules/react-awesome-modal/lib/effect/fadeInRight.js b/node_modules/react-awesome-modal/lib/effect/fadeInRight.js
index d571a2d..013b526 100644
--- a/node_modules/react-awesome-modal/lib/effect/fadeInRight.js
+++ b/node_modules/react-awesome-modal/lib/effect/fadeInRight.js
@@ -40,7 +40,7 @@ exports.default = {
     },
     'panel': {
         'backgroundColor': '#fff',
-        'borderRadius': '5px',
+        'borderRadius': '0px',
         'boxSizing': 'border-box',
         'boxShadow': '0 2px 4px rgba(0, 0, 0, 0.3)',
         'transform': 'translate3d(0, 0, 0)',
diff --git a/node_modules/react-awesome-modal/lib/effect/fadeInUp.js b/node_modules/react-awesome-modal/lib/effect/fadeInUp.js
index d053525..25f8782 100644
--- a/node_modules/react-awesome-modal/lib/effect/fadeInUp.js
+++ b/node_modules/react-awesome-modal/lib/effect/fadeInUp.js
@@ -40,7 +40,7 @@ exports.default = {
     },
     'panel': {
         'backgroundColor': '#fff',
-        'borderRadius': '5px',
+        'borderRadius': '0px',
         'boxSizing': 'border-box',
         'boxShadow': '0 2px 4px rgba(0, 0, 0, 0.3)',
         'transform': 'translate3d(0, 0, 0)',
